### PR TITLE
Overhual of KVM playbooks factoring out vm host networking config

### PR DESCRIPTION
**Pull request by nocturnalastro**

### CODE REVIEW

File: playbooks/configure_vm_host_network.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+- import_playbook: playbooks/process_kvm_nodes.yml
```
Comment (arjuhe) [inline playbooks/configure_vm_host_network.yml:2]: Try adding this to meta/main.yml. You wont need to import the process_kvm_nodes playbook.

```dependencies:
  - { role: process_kvm_nodes, transport_config: ... } ```

File: playbooks/process_kvm_nodes.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+- name: Process KVM nodes
+  hosts: bastion
+  gather_facts: "{{ ((setup_vms | default(true)) and all_kvm_nodes is undefined) | bool }}"
```
Comment (nocturnalastro) [inline playbooks/process_kvm_nodes.yml:3]: Need to fix the gather facts

File: playbooks/destroy_vms.yml [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
-- name: Process KVM nodes
-  hosts: bastion
-  roles:
-    - process_kvm_nodes
+- import_playbook: playbooks/process_kvm_nodes.yml
 
-- name: Destroy VMs
+- name: Destroy VMS
   hosts: vm_hosts
+  gather_facts: "{{ (setup_vms | default((kvm_nodes | default([])) | length | int >= 1)) | bool }}"
```
Comment (nocturnalastro) [inline playbooks/destroy_vms.yml:6]: Need to fix the gather facts

File: playbooks/configure_vm_host_network.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+- name: Configuring vm host network
+  hosts: vm_hosts
+  gather_facts: "{{ configure_vm_host_network | (setup_vms | default((kvm_nodes | default([])) | length | int >= 1)) | bool }}"
```
Comment (nocturnalastro) [inline playbooks/configure_vm_host_network.yml:6]: Need to fix the gather facts

**Comment by nocturnalastro** [comment]
Needs more work PR is stale so will come back to it when have time and remake the PR

**Pull request closed without merging**